src/overlays/actors/ovl_Bg_Jya_1flift/z_bg_jya_1flift.c

```diff
@@ -21,31 +39,162 @@ const ActorInit Bg_Jya_1flift_InitVars = {
+f32 D_80893130[] = { 443.0f, -50.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_1flift/BgJya1flift_Update.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 1200, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 400, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1200, ICHAIN_STOP),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_1flift/BgJya1flift_Draw.s")
+extern UNK_TYPE D_060004A8;
+extern Gfx D_060001F0[];
+
+void BgJya1flift_InitDynapoly(BgJya1flift* this, GlobalContext* globalCtx, void* arg2, s32 moveFlag) {
+    s32 pad;
+    s32 sp30;
+    u32 tempDynaID;
+
+    sp30 = 0;
+    DynaPolyInfo_SetActorMove(&this->dyna, moveFlag);
+    DynaPolyInfo_Alloc(arg2, (void*)&sp30);
+    tempDynaID = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, sp30);
+    this->dyna.dynaPolyId = tempDynaID;
+    if (tempDynaID == 0x32) {
+        // Warning : move BG login failed
+        osSyncPrintf("Warning : move BG 登録失敗(%s %d)(name %d)(arg_data 0x%04x)\n", "../z_bg_jya_1flift.c", 0xB3,
+                     this->dyna.actor.id, this->dyna.actor.params);
+    }
+}
+
+void BgJya1flift_InitCollision(Actor* thisx, GlobalContext* globalCtx) {
+    BgJya1flift* this = THIS;
+
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->dyna.actor, &sCylinderInit);
+    this->dyna.actor.colChkInfo.mass = 0xFF;
+}
+
+void BgJya1flift_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgJya1flift* this = THIS;
+    // 1 F lift
+    osSyncPrintf("(１Ｆリフト)(flag %d)(room %d)\n", (u8)D_808930E0, globalCtx->roomCtx.curRoom.num);
+    this->hasInitialized = 0;
+    if (D_808930E0 != 0) {
+        Actor_Kill(thisx);
+        return;
+    }
+    BgJya1flift_InitDynapoly(this, globalCtx, &D_060004A8, 0);
+    Actor_ProcessInitChain(thisx, sInitChain);
+    BgJya1flift_InitCollision(thisx, globalCtx);
+    if (Flags_GetSwitch(globalCtx, (thisx->params & 0x3F)) != 0) {
+        LINK_AGE_IN_YEARS == YEARS_ADULT ? BgJya1flift_ChangeDirection(this) : func_80892E0C(this);
+    } else {
+        func_80892DB0(this);
+    }
+    thisx->room = -1;
+    D_808930E0 = 1;
+    this->hasInitialized = 1;
+}
+
+void BgJya1flift_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgJya1flift* this = THIS;
+
+    if (this->hasInitialized != 0) {
+        D_808930E0 = 0;
+        Collider_DestroyCylinder(globalCtx, &this->collider);
+        DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+    }
+}
+
+void func_80892DB0(BgJya1flift* this) {
+    this->actionFunc = func_80892DCC;
+    this->dyna.actor.posRot.pos.y = D_80893130[0];
+}
+
+void func_80892DCC(BgJya1flift* this, GlobalContext* globalCtx) {
+    if (Flags_GetSwitch(globalCtx, (this->dyna.actor.params & 0x3F)) != 0) {
```

Comment:
```suggestion
    if (Flags_GetSwitch(globalCtx, (this->dyna.actor.params & 0x3F))) {
```

Reply:
think you mightve missed these louis, github hides a bunch of the review comments and you have to click a button to show them lol